client/wfnews-war/src/main/angular/src/app/components/report-of-fire/report-of-fire.component.ts

```diff
@@ -0,0 +1,204 @@
+import { AfterContentInit, ChangeDetectorRef, Component, ComponentRef, OnInit, ViewChild, ViewContainerRef } from "@angular/core"
+import { RoFTitlePage } from "./title-page/rof-title-page.component";
+import { RoFPermissionsPage } from "./permissions-page/rof-permissions-page.component";
+import { RoFSimpleQuestionPage } from "./simple-question-page/rof-simple-question-page.component";
+import { RoFContactPage } from "./contact-page/rof-contact-page.component";
+import { RoFLocationPage } from "./location-page/rof-location-page.component";
+import { RoFPhotoPage } from "./photo-page/rof-photo-page.component";
+import { ReportOfFire } from "./reportOfFireModel";
+import { RoFComplexQuestionPage } from "./complex-question-page/rof-complex-question-page.component";
+
+import ConfigJson from './report-of-fire.config.json';
+import { RoFCommentsPage } from "./comment-page/rof-comments-page.component";
+import { RoFReviewPage } from "./review-page/rof-review-page.component";
+import { Router } from "@angular/router";
+
+/**
+ * Report of Fire parent container. This will hold the reportOfFire data
+ * (which may move to local storage with the offline support, so a temp thing for now)
+ * As well as the create and store the pages for the "wizard"
+ */
+@Component({
+  selector: 'report-of-fire',
+  templateUrl: './report-of-fire.component.html',
+  styleUrls: ['./report-of-fire.component.scss']
+})
+export class ReportOfFirePage implements OnInit, AfterContentInit {
+  public reportOfFire: ReportOfFire;
+  public pageComponents: Array<ComponentRef<any>> = [];
+  public pageIndex = 0;
+  public allowExit = false;
+  public allowSkip = false;
+  public showProgress = false;
+  public progressSteps = [];
+  public currentStep = 0;
+
+  @ViewChild('dynamic', { static: true, read: ViewContainerRef })
+  private dynamicContainer!: ViewContainerRef;
+
+  constructor(private router: Router, protected cdr: ChangeDetectorRef) {
+    this.pageComponents = [];
+  }
+
+  ngOnInit() {
+    // On RoF form startup, we want to load the config json and iterate through the
```

Comment:
Right now it maps stepped pages via the titles as they all match in the TaC, but I do think it would make sense to change things so each component can track it's next pages via an ID so we don't need to worry about order or anything, we can just keep an array of naviagation events